[lint] add lintrunner support for circleci_linter

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67942
* #67932
* #67936
* #67894
* #67890
* __->__ #67872

As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

suo committed:
[lint] add lintrunner support for circleci_linter

As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

driazati commented on tools/linter/adapters/circleci_linter.py:
```diff
@@ -0,0 +1,153 @@
+README_PATH = os.path.join(PARENT_DIR, "README.md")
+
+
+class LintSeverity(str, Enum):
```
Is there some reason we can't share these between all the lint adapters? One of my main things with the python lint runners right now is they've gotten pretty complicated over time (i.e. we have 1000+ lines to run clang-tidy) so if we do something new it should be understandable / extensible

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]

suo committed:
Update on "[lint] add lintrunner support for circleci_linter"


As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

[ghstack-poisoned]
